technology-considerations/TDR Export Format Requirements.md

```diff
@@ -0,0 +1,426 @@
+contains the actual consignment of files. This is as described by the
+specification.
+
+For this example the consignment is a directory named "OdataReports"
+containing two files (Book1.xslx and Droid Charts.xslx)
+
+The basic bag would look like this.
+
+![](./images/basictdrbag.png)
+
+The TDRConsignmentBag directory requires default tag files to be
+included in the bag.
+
+![](./images/basictdrbagfiles.png)
+
+The bagit.txt file simply details the BagIt version and the encoding
+format.
+
+![](./images/bagittxt.png)
+
+The manifest-sha256.txt file contains the sha256 and the file identifier
+of the consignment files. The path is always referenced from the data
+directory.
+
+![](./images/manifest-sha256.png)
+
+The tagmanifest-sha256.txt file contains the sha256 and the file
+identifier of the tag files.
+
+![](./images/tagmanifest-sha256.png)
+
+The information shown above demonstrates the minimum required to create
+a valid bag. However this may not be sufficient for the needs of TDR if
+further information is required to be transferred along with the files.
+
+Tag files may be used to accomplish the transfer of file level
+information / metadata. Possible suggestions are shown below.
+
+![](./images/bagger.png)
+
+Some of these files may not be required to exist as part of the bag and
+depends on the requirements for TDR. However suggestions are currently
+included here for completeness.
+
+The additional tag files are proposed to record information for a
+specific purpose rather than have information bundled into a single more
+complex file format such as xml. A simple csv format is proposed for
+these files as this may aid with visual inspection of the data once
+received and for simplicity of creation by the TDR application.
+
+Once created any additional tag files must also be included into the
+tagmanifest-\<checksum\> file as shown below.
+
+![](./images/tagmanifest-full.png)
+
+The following section makes some suggestions for these additional files
+and their contents.
+
+### Consignment and Process information
+
+The consignment level information and details of the TDR process are
+proposed to be recorded in the bag-info.txt file. Any properties and
+values added to this file would be the final values of the properties at
+the time of bag creation.
+
+The following list of properties is suggested but can be amended or
+enhanced by the TDR application to fulfil any further requirements that
+may be set.
+
+ | Property  | Description | Example value | Property Type | Possible BagIt Descriptor |
+ |-----------|-------------|---------------|---------------|---------------------------|
+ |Bag-Creator| Name of software and version that created the bag |  TDR 1.0  |Custom  |        
+ |Bag-Size | The size of the complete bag package | 86.2 KB | BagIt           
+ |Bagging-Date | Date the bag was written | 2019-09-09 | BagIt           
+ |Browser | The name and version of the browser | | Custom          
```

Comment:
We spoke about this before, but what should we do in future versions of TDR when a user can add files to an existing consignment, possibly from different browsers? What's the need to record the browser, and would it be better off elsewhere, like in Google Analytics?

Reply:
It's still up for debate if it should be captured at all but it was something requested so I have kept it in. If users used multiple browser for the same consignment then we would need to move the property from Consignment level to file level where the value would be stored in the File metadata. We could move it there now but I kept it there at the moment so that it was visible.

Reply:
Yeah, we can move it into the file-level metadata later. I'm just still curious to know what we need it for, because if it's to understand the users better (rather than for auditing) then it will be easier to analyse in something like GA.

Reply:
The request was made to include it by DC so we could analyse the context of the users environment.

Reply:
Anything where _we_ are using some tool to assert something about the files within a consignment is potentially of provenance value. In this case, the browser, version, and implicit in that its relative support of FileAPI and other client-side libraries can be a useful thing to record. We care at the point of the thing being asserted, so if they use one browser to upload (and therefore initiate client-side processes) and a second one to confirm the transfer at the end, I'm most interested in the first part.